### PR TITLE
feat(nuxt): allow keeping fallback for `NuxtClientFallback`

### LINKED ISSUE #20315: Provide a `strategy` prop for `NuxtClientFallback`
### Describe the feature

We could improve the component by adding a `strategy` prop to keep the fallback content once in client-side. Like `default` and `keep-fallback`

### Additional information

- [X] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://nuxt.com/docs/community/contribution).
- [X] Check existing [discussions](https://github.com/nuxt/nuxt/discussions) and [issues](https://github.com/nuxt/nuxt/issues).

### PR DESCRIPTION
**Pull request by huang-julien**
### 🔗 Linked issue
resolve #20315 

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Hi :wave: 
This PR introduce a small feature to `NuxtClientFallback` to keep the fallback content once in client side if it failed in SSR

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: docs/3.api/2.components/1.nuxt-client-fallback.md [hunk 1]
```diff
@@ -30,6 +30,9 @@ This component is experimental and in order to use it you must enable the `exper
   - **default**: `div`
 - **placeholder** | **fallback**: Specify fallback content to be rendered if the slot fails to render.
   - **type**: `string`
+- **strategy**: Specify a strategy to adopt once in client-side. By default it tries to render the slot if it fails in client-side but your can choose to keep the fallback content by setting this prop to `keep-fallback`
+  - **type**: `string`
```
Comment (florian-lefebvre) [inline docs/3.api/2.components/1.nuxt-client-fallback.md:34]: Maybe we could say something like 
```
`default | keep-fallback`
```

File: docs/3.api/2.components/1.nuxt-client-fallback.md [hunk 2]
```diff
@@ -30,6 +30,9 @@ This component is experimental and in order to use it you must enable the `exper
   - **default**: `div`
 - **placeholder** | **fallback**: Specify fallback content to be rendered if the slot fails to render.
```
[the remark this answers is not in the document]
Comment (florian-lefebvre) [reply docs/3.api/2.components/1.nuxt-client-fallback.md:31]: My bad I confused the props names

File: docs/3.api/2.components/1.nuxt-client-fallback.md [hunk 3]
```diff
@@ -30,6 +30,9 @@ This component is experimental and in order to use it you must enable the `exper
...
   - **type**: `string`
+- **strategy**: Specify a strategy to adopt once in client-side. By default it tries to render the slot if it fails in client-side but your can choose to keep the fallback content by setting this prop to `keep-fallback`
```
Comment (danielroe) [inline docs/3.api/2.components/1.nuxt-client-fallback.md:33]:
```suggestion
- **strategy**: Specify a strategy to adopt once client-side. By default it tries to re-render the slot on the client but you can choose to keep the fallback content by setting this prop to `keep-fallback`
```

File: docs/3.api/2.components/1.nuxt-client-fallback.md [hunk 4]
```diff
@@ -30,6 +30,9 @@ This component is experimental and in order to use it you must enable the `exper
...
+  - **type**: `'default'|'fallback'`
```
Comment (danielroe) [inline docs/3.api/2.components/1.nuxt-client-fallback.md:34]:
```suggestion
  - **type**: `'default' | 'fallback'`
```

File: packages/nuxt/src/app/components/client-fallback.client.ts [hunk 5]
```diff
@@ -21,6 +21,10 @@ export default defineComponent({
     },
     placeholderTag: {
       type: String
+    },
+    strategy: {
+      type: String as () => 'default'|'keep-fallback',
```
Comment (danielroe) [inline packages/nuxt/src/app/components/client-fallback.client.ts:26]:
```suggestion
      type: String as () => 'default' | 'keep-fallback',
```

**Review by danielroe — APPROVED** [review]

**Comment by huang-julien** [comment]
@danielroe I've implemented this as a string prop. Do you think we should make it a boolean ?

**Comment by danielroe** [comment]
I'd also be happy with that 👍

**Comment by huang-julien** [comment]
Great ! i'll update this tonight. There's seems to be an issue with webpack since the branch update -> will take a look and fix it 👀

**Comment by huang-julien** [comment]
@danielroe i'm not sure when, but an update completly broke the component when using webpack 🤔

**Comment by danielroe** [comment]
Hm. Maybe we could try locally bisecting the branch to see which it was?

**Comment by huang-julien** [comment]
Found it, will open an issue + PR 🤔 not sure why it break when we have a `transformInclude` with `vue&type=template`

**Pull request merged**